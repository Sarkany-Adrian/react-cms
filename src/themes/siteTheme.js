// @flow
// shared declarations
import { font, baseColors } from './shared';
// theme specific declarations
import theme1Colors from './theme1';
import theme2Colors from './theme2';
import theme3Colors from './theme3';

const siteTheme = {
  theme1: {
    id: 1,
    colors: theme1Colors,
    font,
    baseColors
  },
  theme2: {
    id: 2,
    colors: theme2Colors,
    font,
    baseColors
  },
  theme3: {
    id: 3,
    colors: theme3Colors,
    font,
    baseColors
  }
};

export default siteTheme;
