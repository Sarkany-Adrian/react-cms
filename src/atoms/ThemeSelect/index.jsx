// @flow
import React from 'react';
// theme
import { ThemeProvider } from 'styled-components';
import { SiteThemeContext } from 'contexts/theme';
import siteTheme from 'themes/siteTheme';

type TProps = {
  handleThemeChange?: (e: any) => void,
  theme?: Object
};

function ThemeSelect(): React$Element<*> {
  return (
    <SiteThemeContext.Consumer>
      {(props: ?TProps) => {
        if (props) {
          const { handleThemeChange, theme } = props;
          return (
            <ThemeProvider theme={theme}>
              <select
                onChange={(e: SyntheticMouseEvent<*>) =>
                  handleThemeChange && handleThemeChange(e)
                }
              >
                {Object.keys(siteTheme).map((themeName: any, index: number) => (
                  <option key={themeName} value={themeName}>
                    Theme {index + 1}
                  </option>
                ))}
              </select>
            </ThemeProvider>
          );
        }
        return null;
      }}
    </SiteThemeContext.Consumer>
  );
}

export default ThemeSelect;
