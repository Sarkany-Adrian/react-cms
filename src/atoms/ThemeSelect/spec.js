import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import toJson from 'enzyme-to-json';
import ThemeSelect from 'atoms/ThemeSelect';
import siteTheme from 'themes/siteTheme';

jest.mock('contexts/theme', () => {
  const context = {
    theme: { name: 'test_theme' },
    handleThemeChange: jest.fn()
  };

  console.log('here', context);

  return {
    SiteThemeContext: {
      Consumer: props => props.children(context)
    }
  };
});

describe('Theme Select Atom', () => {
  // const shallowWithTheme = (tree, theme) => {
  //   const context = shallow(<ThemeProvider theme={theme} />)
  //     .instance()
  //     .getChildContext();
  //   return shallow(tree, { context });
  // };

  const context = {
    theme: siteTheme.theme1
  };

  beforeEach(() => {
    jest.resetModules();
  });

  it('should render and match snapshot', () => {
    const wrapper = shallow(<ThemeSelect />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should trigger handleThemeChange on select', () => {
    // const outer = mount(<ThemeSelect />);

    // const Children = outer.props().children({ theme });
    // const wrapper = shallow(Children);
    //
    // wrapper
    //   .dive()
    //   .find('select')
    //   .simulate('change');
    //
    // expect(handleThemeChange).toHaveBeenCalled();

    // const wrapper = shallowWithTheme(<ThemeSelect />, theme);
    //
    // console.log(wrapper.debug());
    // const ThemeSelectComp = getLanguageSelectorWithContext();
    const wrapper = mount(
      <ThemeProvider
        theme={context.theme}
        handleThemeChange={context.handleThemeChange}
      >
        <ThemeSelect />
      </ThemeProvider>
    );

    const select = wrapper.find('select');

    console.log(select.debug());

    select.simulate('change');

    console.log('longshot', context);

    expect(context.handleThemeChange).toHaveBeenCalled();

    // expect(select).toHaveLength(1);
  });
});
