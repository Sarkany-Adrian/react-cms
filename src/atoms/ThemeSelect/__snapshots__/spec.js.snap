// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Select Atom should render and match snapshot 1`] = `
<Consumer>
  <Component />
</Consumer>
`;
